File: 0089-gray-code/0089-gray-code.py
class Solution:
    def grayCode(self, n: int):
        result = [0]
        for i in range (1, n+1):
            prev = len(result)
            mask = 1 << (i - 1)
            for j in range(prev - 1, -1, -1):
                result.append(mask + result[j])
        return result
        

File: 0089-gray-code/0089-gray-code.java
class Solution {
    public List<Integer> grayCode(int n) {
        int length = 1<<n;
        List<Integer> ans = new ArrayList<>();

        for (int i = 0; i < length; i++) {
            ans.add(i ^ (i >> 1));
        }

        return ans;
    }
}